File: src/main/java/me/miki/shindo/management/notification/NotificationHandler.java
package me.miki.shindo.management.notification;

import me.miki.shindo.management.event.EventTarget;
import me.miki.shindo.management.event.impl.EventRenderNotification;

import java.util.concurrent.LinkedBlockingQueue;

public class NotificationHandler {

    private final LinkedBlockingQueue<Notification> notifications;
    private Notification currentNotification;

    public NotificationHandler(LinkedBlockingQueue<Notification> notifications) {
        this.notifications = notifications;
    }

    @EventTarget
    public void onRenderNotification(EventRenderNotification event) {

        if (currentNotification != null && !currentNotification.isShown()) {
            currentNotification = null;
        }

        if (currentNotification == null && !notifications.isEmpty()) {
            currentNotification = notifications.poll();
            currentNotification.show();
        }

        if (currentNotification != null) {
            currentNotification.draw();
        }
    }
}

File: src/main/kotlin/me/miki/shindo/management/notification/NotificationHandler.kt
package me.miki.shindo.management.notification

import me.miki.shindo.management.event.EventTarget
import me.miki.shindo.management.event.impl.EventRenderNotification
import java.util.concurrent.LinkedBlockingQueue

class NotificationHandler(private val notifications: LinkedBlockingQueue<Notification>) {

    private var currentNotification: Notification? = null

    @EventTarget
    fun onRenderNotification(event: EventRenderNotification) {
        if (currentNotification != null && currentNotification?.isShown() == false) {
            currentNotification = null
        }

        if (currentNotification == null && notifications.isNotEmpty()) {
            currentNotification = notifications.poll()
            currentNotification?.show()
        }

        currentNotification?.draw()
    }
}
